File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let sumMap = new Map();
    for(let i=0;i<nums.length;i++){
        const diff = target - nums[i];
        if(sumMap.has(diff)){
            return [i,sumMap.get(diff)];
        }
        else{
            sumMap.set(nums[i],i);
        }
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        sumMap = {}
        diff = 0
        for i in range(len(nums)) :
            diff = target - nums[i]
            if diff in sumMap :
                return [i,sumMap[diff]]
            else :
                sumMap[nums[i]] = i
        return -1